R/labelled.R

```diff
@@ -0,0 +1,208 @@
+#' @title
+#' add labels!!!
+#'
+#' @param supertbl a supertibble containing REDCap data created by
+#' \code{read_redcap_tidy()}
+#' @param format_labels One of:
+#' \itemize{
+#'   \item \code{NULL} to apply field labels to elements of \code{redcap_data}
+#'   as they appear in \code{redcap_metadata}
+#'   \item The name of a \code{REDCapTidieR} \code{label_format_*} function.
+#'   Currently only \code{"default"}
+#'   \item A function that takes the labels in \code{redcap_metadata} as input
+#'   and returns a vector of formatted labels of the same length as output
+#' }
+#'
+#' @importFrom dplyr %>%
+#' @importFrom purrr map map2
+#' @importFrom rlang check_installed
+#' @importFrom checkmate assert_data_frame
+#'
+#' @return
+#' Supertibble with labels applied to:
+#' - x
+#' - y
+#' - z
+#'
+#' @examples
+#' supertbl <- tibble::tribble(
+#'   ~ redcap_data, ~ redcap_metadata,
+#'   tibble::tibble(x = letters[1:3]), tibble::tibble(field_name = "x", field_label = "X Label"),
+#'   tibble::tibble(y = letters[1:3]), tibble::tibble(field_name = "y", field_label = "Y Label")
+#' )
+#'
+#' make_labelled(supertbl)
+#'
+#' \dontrun{
+#' redcap_uri <- Sys.getenv("REDCAP_URI")
+#' token <- Sys.getenv("REDCAP_TOKEN")
+#'
+#' supertbl <- read_redcap_tidy(redcap_uri, token)
+#' make_labelled(supertbl)
+#' }
+#' @export
+make_labelled <- function(supertbl, format_labels = NULL) {
+
+  check_installed("labelled", reason = "to use `make_labelled()`")
+
+  if (is.null(format_labels)) {
+    formatter <- identity
+  } else if (is.function(format_labels)) {
+    formatter <- format_labels
+  } else if (format_labels == "default") {
+    formatter <- format_labels_default
+  } else {
+    # TODO: add informative error with cli
+    stop()
```

Comment:
To immediate mind we'd want a catch like:

```
"✖ ERROR: argument `formatter` must be of type 'function'" 
ℹ "Or use "default" for `REDCapTidieR` default formatting options".
```
Or something better worded to point users to default settings. I can't think of much else we would check for here, if a custom formatter fails it's mostly the user's responsibility.

Reply:
Yeah something like this makes sense to me. I'll add it when I sure up the interface and documentation